source/common/grpc/async_client_impl.h

```diff
@@ -49,8 +49,19 @@ class AsyncClientImpl final : public RawAsyncClient {
   friend class AsyncStreamImpl;
 };
 
+template <typename T> class CarryOn {
+public:
+  T& carryOn() { return carry_on_; }
+  void setCarryOn(T&& carry_on) { carry_on_ = std::move(carry_on); }
```

Comment:
[q] Should we add the ability to support multiple callbacks in a list (i.e. `setCarryOn` simply appends the unique pointer to the callback to some internal list)?

Reply:
Each stream is associated with only one callback, so we don't need multiple callbacks.

Reply:
Sorry, maybe I'm misunderstanding but why is this a limitation?

Reply:
Since the function to start a stream only accepts one callback. I don’t think it’s a limitation though. Why would multiple callbacks needed?

Reply:
if we do decide that this is useful in the future, i guess we could just have a wrapper callback that we pass that has it's own embedded callbacks :-)